### PR TITLE
chore: update subscriptions docs to include multipart over HTTP

### PR DESCRIPTION
**Pull request by alessbell**

### CODE REVIEW

File: docs/source/data/subscriptions.mdx [hunk 1]
```diff
@@ -26,7 +26,15 @@ You _should_ use subscriptions for the following:
 
 ## Choosing a subscription library
 
-The GraphQL spec does not define a specific protocol for sending subscription requests. The first popular JavaScript library to implement subscriptions over WebSocket is called `subscriptions-transport-ws`. **This library is no longer actively maintained.** Its successor is a library called `graphql-ws`. The two libraries _do not use the same WebSocket subprotocol_, so you need to make sure that your server and clients all use the same library.
+The GraphQL spec does not define a specific protocol for sending subscription requests. In Apollo Client, subscriptions can be sent over HTTP via multipart transport with the default terminating link, `HTTPLink`, or via WebSocket.
+
+### Multipart Subscriptions Over HTTP
+
+In order to use Apollo Client with a back-end that supports multipart subscriptions over HTTP (for example, Apollo Router) no additional client configuration is required. Apollo Client will automatically send the required headers with the request if the terminating `HTTPLink` receives a subscription operation.
```
Comment (alessbell) [inline docs/source/data/subscriptions.mdx:33]: Should we list/link to other docs here besides/in addition to the router?

File: docs/source/data/subscriptions.mdx [hunk 2]
```diff
@@ -26,7 +26,15 @@ You _should_ use subscriptions for the following:
 
 ## Choosing a subscription library
 
-The GraphQL spec does not define a specific protocol for sending subscription requests. The first popular JavaScript library to implement subscriptions over WebSocket is called `subscriptions-transport-ws`. **This library is no longer actively maintained.** Its successor is a library called `graphql-ws`. The two libraries _do not use the same WebSocket subprotocol_, so you need to make sure that your server and clients all use the same library.
+The GraphQL spec does not define a specific protocol for sending subscription requests. In Apollo Client, subscriptions can be sent over HTTP via multipart transport with the default terminating link, `HTTPLink`, or via WebSocket.
+
+### Multipart Subscriptions Over HTTP
```
Comment (alessbell) [inline docs/source/data/subscriptions.mdx:31]: > Multipart Subscriptions

Review for naming conventions/consistency

File: docs/source/data/subscriptions.mdx [hunk 3]
```diff
@@ -80,9 +88,15 @@ Whenever your GraphQL server _does_ push data to a subscribing client, that data
 
 ## Setting up the transport
 
-Because subscriptions usually maintain a persistent connection, they shouldn't use the default HTTP transport that Apollo Client uses for queries and mutations. Instead, Apollo Client subscriptions most commonly communicate over WebSocket, via the [`graphql-ws`](https://www.npmjs.com/package/graphql-ws) library.
+Additional configuration is required if the back-end service your application uses communicates over the WebSocket protocol; in this case, Apollo Client subscriptions most commonly use the [`graphql-ws`](https://www.npmjs.com/package/graphql-ws) library. If the back-end service supports multipart subscriptions via HTTP, however, _no additional configuration is required_.
```
Comment (alessbell) [inline docs/source/data/subscriptions.mdx:91]:
```suggestion
Additional configuration is required if the back-end service your application communicates with is using the WebSocket protocol for subscriptions; in this case, Apollo Client most commonly use the [`graphql-ws`](https://www.npmjs.com/package/graphql-ws) library to provide WebSocket support. If the back-end service supports subscriptions over multipart HTTP, however, _no additional configuration is required_.
```

File: docs/source/development-testing/static-typing.md [hunk 4]
```diff
@@ -256,7 +256,7 @@ export function NewRocketForm() {
 
 ### `useSubscription`
 
-We can type our `useSubscription` hooks the same way we typed our `useQuery` and ` useMutation` hooks. Using the generated `gql` function to define our GraphQL subscriptions, we ensure that we type our subscription variables and return data:
```
Comment (alessbell) [inline docs/source/development-testing/static-typing.md:259]: Removed unnecessary whitespace

File: docs/source/data/subscriptions.mdx [hunk 5]
```diff
@@ -80,9 +88,15 @@ Whenever your GraphQL server _does_ push data to a subscribing client, that data
...
+
+> As mentioned in [Choosing a subscription library](#choosing-a-subscription-library), some servers that communicate via WebSocket use an older library called `subscriptions-transport-ws`. For necessary changes to use that library, see [below](#the-older-subscriptions-transport-ws-library).
+
+## Subscriptions via Multipart HTTP
+
+No additional libraries or configuration are required. Apollo Client add the required headers to your request when the terminating (and default) `HTTPLink` receives a subscription operation at the `uri` specified when initializing the link or Apollo Client instance.
```
Comment (jerelmiller) [inline docs/source/data/subscriptions.mdx:97]:
```suggestion
No additional libraries or configuration are required. Apollo Client adds the required headers to your request when the terminating (and default) `HTTPLink` receives a subscription operation at the `uri` specified when initializing the link or Apollo Client instance.
```

Comment (jerelmiller) [inline docs/source/data/subscriptions.mdx:97]:
```suggestion
No additional libraries or configuration are required. Apollo Client add the required headers to your request when the default terminating `HTTPLink` receives a subscription operation at the `uri` specified when initializing the link or Apollo Client instance.
```

I think removing the parens and moving "default" before "terminating" makes it read a bit better. Let me know what you think.

File: docs/source/data/subscriptions.mdx [hunk 6]
```diff
@@ -80,9 +88,15 @@ Whenever your GraphQL server _does_ push data to a subscribing client, that data
 
 ## Setting up the transport
 
-Because subscriptions usually maintain a persistent connection, they shouldn't use the default HTTP transport that Apollo Client uses for queries and mutations. Instead, Apollo Client subscriptions most commonly communicate over WebSocket, via the [`graphql-ws`](https://www.npmjs.com/package/graphql-ws) library.
+Additional configuration is required if the back-end service your application uses communicates over the WebSocket protocol; in this case, Apollo Client subscriptions most commonly use the [`graphql-ws`](https://www.npmjs.com/package/graphql-ws) library. If the back-end service supports multipart subscriptions via HTTP, however, _no additional configuration is required_.
+
+> As mentioned in [Choosing a subscription library](#choosing-a-subscription-library), some servers that communicate via WebSocket use an older library called `subscriptions-transport-ws`. For necessary changes to use that library, see [below](#the-older-subscriptions-transport-ws-library).
+
+## Subscriptions via Multipart HTTP
```
Comment (jerelmiller) [inline docs/source/data/subscriptions.mdx:95]: Some of this feels a bit redundant to the "Multipart Subscriptions Over HTTP" section (same with the websockets). Would it make sense to try and combine these sections together?

File: docs/source/data/subscriptions.mdx [hunk 7]
```diff
@@ -26,7 +26,15 @@ You _should_ use subscriptions for the following:
 
 ## Choosing a subscription library
 
-The GraphQL spec does not define a specific protocol for sending subscription requests. The first popular JavaScript library to implement subscriptions over WebSocket is called `subscriptions-transport-ws`. **This library is no longer actively maintained.** Its successor is a library called `graphql-ws`. The two libraries _do not use the same WebSocket subprotocol_, so you need to make sure that your server and clients all use the same library.
+The GraphQL spec does not define a specific protocol for sending subscription requests. In Apollo Client, subscriptions can be sent over HTTP via multipart transport with the default terminating link, `HTTPLink`, or via WebSocket.
```
Comment (StephenBarlow) [inline docs/source/data/subscriptions.mdx:29]:
```suggestion
The GraphQL spec does not define a specific protocol for sending subscription requests. Apollo Client supports the following protocols for subscriptions:

- WebSocket, using one of the following subprotocols:
    - [`graphql-ws`](https://github.com/enisdenjo/graphql-ws)
    - [`subscriptions-transport-ws`](https://github.com/apollographql/subscriptions-transport-ws) (⚠️ unmaintained)
- HTTP, using [chunked multipart responses](https://github.com/graphql/graphql-over-http/blob/main/rfcs/IncrementalDelivery.md) **(Apollo Client x.y.z and later)**

You must use the same protocol as the GraphQL endpoint you're communicating with.
```

File: docs/source/data/subscriptions.mdx [hunk 8]
```diff
@@ -26,7 +26,15 @@ You _should_ use subscriptions for the following:
 
 ## Choosing a subscription library
```
Comment (StephenBarlow) [inline docs/source/data/subscriptions.mdx:27]:
```suggestion
## Supported subscription protocols
```

File: docs/source/data/subscriptions.mdx [hunk 9]
```diff
@@ -26,7 +26,15 @@ You _should_ use subscriptions for the following:
...
 
-The GraphQL spec does not define a specific protocol for sending subscription requests. The first popular JavaScript library to implement subscriptions over WebSocket is called `subscriptions-transport-ws`. **This library is no longer actively maintained.** Its successor is a library called `graphql-ws`. The two libraries _do not use the same WebSocket subprotocol_, so you need to make sure that your server and clients all use the same library.
+The GraphQL spec does not define a specific protocol for sending subscription requests. In Apollo Client, subscriptions can be sent over HTTP via multipart transport with the default terminating link, `HTTPLink`, or via WebSocket.
+
+### Multipart Subscriptions Over HTTP
+
+In order to use Apollo Client with a back-end that supports multipart subscriptions over HTTP (for example, Apollo Router) no additional client configuration is required. Apollo Client will automatically send the required headers with the request if the terminating `HTTPLink` receives a subscription operation.
+
+### WebSocket
```
Comment (StephenBarlow) [inline docs/source/data/subscriptions.mdx:35]:
```suggestion
### WebSocket subprotocols
```

File: docs/source/data/subscriptions.mdx (hunk 2 above)
Comment (StephenBarlow) [inline docs/source/data/subscriptions.mdx:31]:
```suggestion
### HTTP
```

Comment (StephenBarlow) [reply docs/source/data/subscriptions.mdx:31]: Recommend keeping it short and expanding in the body text, and also moving this subsection _below_ the WebSocket subsection

File: docs/source/data/subscriptions.mdx (hunk 1 above)
Comment (StephenBarlow) [inline docs/source/data/subscriptions.mdx:33]:
```suggestion
To use Apollo Client with a backend that supports multipart subscriptions over HTTP, no additional client configuration is required. Apollo Client automatically sends the required headers with the request if the terminating `HTTPLink` is passed a subscription operation.
```

File: docs/source/data/subscriptions.mdx [hunk 10]
```diff
@@ -80,9 +88,15 @@ Whenever your GraphQL server _does_ push data to a subscribing client, that data
 
 ## Setting up the transport
 
-Because subscriptions usually maintain a persistent connection, they shouldn't use the default HTTP transport that Apollo Client uses for queries and mutations. Instead, Apollo Client subscriptions most commonly communicate over WebSocket, via the [`graphql-ws`](https://www.npmjs.com/package/graphql-ws) library.
+Additional configuration is required if the back-end service your application uses communicates over the WebSocket protocol; in this case, Apollo Client subscriptions most commonly use the [`graphql-ws`](https://www.npmjs.com/package/graphql-ws) library. If the back-end service supports multipart subscriptions via HTTP, however, _no additional configuration is required_.
+
+> As mentioned in [Choosing a subscription library](#choosing-a-subscription-library), some servers that communicate via WebSocket use an older library called `subscriptions-transport-ws`. For necessary changes to use that library, see [below](#the-older-subscriptions-transport-ws-library).
```
Comment (StephenBarlow) [inline docs/source/data/subscriptions.mdx:93]:
```suggestion
> As mentioned in [Supported subscription protocols](#supported-subscription-protocols), some GraphQL servers use an older WebSocket library called `subscriptions-transport-ws`. For necessary changes to use that library, see [below](#the-older-subscriptions-transport-ws-library).
```

File: docs/source/data/subscriptions.mdx [hunk 11]
```diff
@@ -80,9 +88,15 @@ Whenever your GraphQL server _does_ push data to a subscribing client, that data
...
+
+## Subscriptions via Multipart HTTP
+
+No additional libraries or configuration are required. Apollo Client adds the required headers to your request when the terminating (and default) `HTTPLink` receives a subscription operation at the `uri` specified when initializing the link or Apollo Client instance.
 
-> As mentioned in [Choosing a subscription library](#choosing-a-subscription-library), some servers use an older library called `subscriptions-transport-ws`. For necessary changes to use that library, see [below](#the-older-subscriptions-transport-ws-library).
+## Subscriptions via WebSocket
```
Comment (StephenBarlow) [inline docs/source/data/subscriptions.mdx:99]:
```suggestion
## WebSocket setup
```

**Review by StephenBarlow — COMMENTED** [review]
Looking good, some initial suggestions!

File: docs/source/data/subscriptions.mdx [hunk 12]
```diff
@@ -26,7 +26,15 @@ You _should_ use subscriptions for the following:
...
 ## Choosing a subscription library
 
-The GraphQL spec does not define a specific protocol for sending subscription requests. The first popular JavaScript library to implement subscriptions over WebSocket is called `subscriptions-transport-ws`. **This library is no longer actively maintained.** Its successor is a library called `graphql-ws`. The two libraries _do not use the same WebSocket subprotocol_, so you need to make sure that your server and clients all use the same library.
+The GraphQL spec does not define a specific protocol for sending subscription requests. In Apollo Client, subscriptions can be sent over HTTP via multipart transport with the default terminating link, `HTTPLink`, or via WebSocket.
+
+### Multipart Subscriptions Over HTTP
+
+To use Apollo Client with a backend that supports multipart subscriptions over HTTP, no additional client configuration is required. Apollo Client automatically sends the required headers with the request if the terminating `HTTPLink` is passed a subscription operation.
+
+### WebSocket
```
Comment (StephenBarlow) [inline docs/source/data/subscriptions.mdx:35]:
```suggestion
### WebSocket subprotocols
```

File: docs/source/data/subscriptions.mdx (hunk 5 above)
Comment (alessbell) [reply docs/source/data/subscriptions.mdx]: I like it, thanks! Applying it manually since it no longer will allow me to commit it here on an outdated diff 👍

File: docs/source/data/subscriptions.mdx [hunk 13]
```diff
@@ -24,14 +24,29 @@ You _should_ use subscriptions for the following:
...
+- WebSocket, using one of the following subprotocols:
+    - [`graphql-ws`](https://github.com/enisdenjo/graphql-ws)
+    - [`subscriptions-transport-ws`](https://github.com/apollographql/subscriptions-transport-ws) (⚠️ unmaintained)
+- HTTP, using [chunked multipart responses](https://github.com/graphql/graphql-over-http/blob/main/rfcs/IncrementalDelivery.md) **(Apollo Client `3.7.11` and later)**
+
+You must use the same protocol as the GraphQL endpoint you're communicating with.
+
+### WebSocket subprotocols
+
+The first popular JavaScript library to implement subscriptions over WebSocket is called `subscriptions-transport-ws`. **This library is no longer actively maintained.** Its successor is a library called `graphql-ws`. The two libraries _do not use the same WebSocket subprotocol_, so you need to make sure that your server and clients all use the same library.
```
Comment (StephenBarlow) [inline docs/source/data/subscriptions.mdx:40]:
```suggestion
The first popular JavaScript library to implement subscriptions over WebSocket is called `subscriptions-transport-ws`. **This library is no longer actively maintained.** Its successor is a library called `graphql-ws`. These two libraries _do not use the same WebSocket subprotocol_, so you need to use the same subprotocol that your GraphQL endpoint uses.
```

File: docs/source/data/subscriptions.mdx [hunk 14]
```diff
@@ -24,14 +24,29 @@ You _should_ use subscriptions for the following:
...
 
 Apollo Client supports both `graphql-ws` and `subscriptions-transport-ws`. We recommend you use the newer library `graphql-ws`, and this page shows how to use it. If you need to use `subscriptions-transport-ws` because your server still uses that protocol, the differences are described [at the bottom of this page](#the-older-subscriptions-transport-ws-library).
```
Comment (StephenBarlow) [inline docs/source/data/subscriptions.mdx:42]:
```suggestion
The [WebSocket setup](#websocket-setup) section below uses `graphql-ws`. If your endpoint uses `subscriptions-transport-ws`, see [this section](#the-older-subscriptions-transport-ws-library) for differences in configuration.
```

File: docs/source/data/subscriptions.mdx [hunk 15]
```diff
@@ -24,14 +24,29 @@ You _should_ use subscriptions for the following:
...
 
 > **Note**: Confusingly, the `subscriptions-transport-ws` library calls its _WebSocket subprotocol_ `graphql-ws`, and the `graphql-ws` _library_ calls its subprotocol `graphql-transport-ws`! In this article, we refer to the two _libraries_ (`subscriptions-transport-ws` and `graphql-ws`), _not_ the two subprotocols.
 
+### HTTP
+
+To use Apollo Client with a backend that supports multipart subscriptions over HTTP, no additional client configuration is required. Apollo Client automatically sends the required headers with the request if the terminating `HTTPLink` is passed a subscription operation.
```
Comment (StephenBarlow) [inline docs/source/data/subscriptions.mdx:48]:
```suggestion
To use Apollo Client with a GraphQL endpoint that supports [multipart subscriptions over HTTP](https://github.com/graphql/graphql-over-http/blob/main/rfcs/IncrementalDelivery.md), make sure you're using version `3.7.11` or later.

Aside from updating your client version, no additional configuration is required! Apollo Client automatically sends the required headers with the request if the terminating `HTTPLink` is passed a subscription operation.
```

File: docs/source/data/subscriptions.mdx (hunk 6 above)
Comment (StephenBarlow) [reply docs/source/data/subscriptions.mdx:102]: Yeah @alessbell I recommend cutting the "Setting up the transport" header, all the way down to line 104 below. And put the callout about `subscriptions-transport-ws`  just below the `WebSocket setup` header

**Review by StephenBarlow — APPROVED** [review]
Few more things but looking great! Handy that it's so easy to use 😄

File: docs/source/data/subscriptions.mdx (hunk 6 above)
Comment (alessbell) [reply docs/source/data/subscriptions.mdx]: Thanks for the feedback here! Fixed in https://github.com/apollographql/apollo-client/pull/10663/commits/0e7cfbc87694b60b117ac2608b65a88592fa6457.

**Comment by alessbell** [comment]
Thanks @StephenBarlow! I've applied all of your suggestions, really appreciate the thorough review 🙇‍♀️

**Review by hwillson — APPROVED** [review]
This is awesome - excited to get this published soon. Thanks for all of your work on this @alessbell!

**Review by phryneas — APPROVED** [review]

**Review by jerelmiller — APPROVED** [review]
🔥 🔥

**Comment by alessbell** [comment]
Hi @enisdenjo 👋

You’re right, HTTP chunked transfer encoding is obsolete in HTTP/2. I’ve updated the PR to remove the reference to the Incremental Delivery over HTTP RFC, since it seems the language there was causing the confusion.

Apollo Client supports subscriptions over HTTP using our own Apollo protocol which uses a multipart/mixed response format similar to the one specified in the Incremental Delivery RFC, but does not use chunked transfer encoding in HTTP/2 (only when the request is sent over HTTP/1.1) and is fully HTTP/2 compatible. Appreciate the feedback!

**Comment by enisdenjo** [comment]
Just wanted to give you guys a heads up that with https://github.com/graphql/graphql-over-http/pull/252, GraphQL's Incremental Delivery spec describes how HTTP/2 enabled servers should behave in greater detail.
